### PR TITLE
Support numpy/python 32 bit float type names

### PR DESCRIPTION
**Pull request by hellkite500**
Add support for numpy/python 32 bit float type names in python bmi adapter.

## Changes

- python adapter now maps `float32`, `np.float32`, and `numpy.float32` to the cxx `float` type

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows project standards (link if applicable)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [x] Linux
- [x] MacOs

### CODE REVIEW

File: include/bmi/Bmi_Py_Adapter.hpp [hunk 1]
```diff
@@ -248,7 +248,8 @@ namespace models {
                     return "long long";
                 } else if (py_type_name == "longlong" && item_size == sizeof(long long)) {
                     return "long long"; //numpy type
-                } else if (py_type_name == "float" && item_size == sizeof(float)) {
+                } else if ( (py_type_name == "float" || py_type_name == "float32" || py_type_name == "np.float32" ||
+                           py_type_name == "numpy.float32") && item_size == sizeof(float)) {
```
Comment (robertbartel) [inline include/bmi/Bmi_Py_Adapter.hpp:252]: We might need to account for this as [`numpy.single`](https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.single) as well: 

```suggestion
                           py_type_name == "numpy.float32" || py_type_name == "np.single" ||  py_type_name == "numpy.single") && item_size == sizeof(float)) {
```

**Review by robertbartel — CHANGES_REQUESTED** [review]
I had to go look it up, but apparently numpy's single-precision float type goes by the name `numpy.single` as well.  I suggested a change to account for this too.

Regardless, should we also update the _Variable Types and Sizes_ section of _doc/BMIconventions.md_?

**Comment by hellkite500** [comment]
Updated bmi docs in 1d22bbcca4f3264f491f00580b09851df4eeaa41

**Review by robertbartel — APPROVED** [review]

File: doc/BMIconventions.md [hunk 2]
```diff
@@ -81,7 +81,19 @@ Since the [BMI Documentation] simply states that, "Use of native language type n
     * `get_var_itemsize` result **must** match the size of the data type in the compiler used to build ngen
 * Python
     * `int`, `long`, `long long`, `int64`, `longlong`, `float`, `float64`, `long double`
```
Comment (PhilMiller) [inline doc/BMIconventions.md:83]: This missed documenting that `float32` is accepted

Comment (PhilMiller) [reply doc/BMIconventions.md:83]: (without a `numpy.` or `np.` prefix)

**Pull request merged**